Source/WebCore/page/SecurityOriginData.h

```diff
@@ -95,14 +116,24 @@ struct SecurityOriginData {
 #if !LOG_DISABLED
     String debugString() const { return toString(); }
 #endif
+private:
+    friend struct IPC::ArgumentCoder<SecurityOriginData, void>;
+    SecurityOriginData(const String& protocol, const String& host, std::optional<uint16_t> port, MarkableOpaqueOriginIdentifier opaqueOriginIdentifier)
```

Comment:
This constructor is overly dangerous and should probably not exist. When IPC comes from the WebProcess, it cannot be trusted, this means that IPC from a compromised WebProcess could construct a SecurityOriginData object that is in a weird state (and we could lead to bad behavior or crashes). By weird state, I mean for example that opaqueOriginIdentifier could be non-empty and yet we could have a valid-looking protocol / host.

My personal preference would be to drop this constructor and write our own encode / decode functions that would look like so:
```
encode()
{
    encoder << opaqueOriginIdentifier;
    if (opaqueOriginIdentifier)
        return;
   encoder << protocol << host << port;
}

decode()
{
    std::optional<MarkableOpaqueOriginIdentifier> opaqueOriginIdentifier;
    decoder >> opaqueOriginIdentifier;
    if (!opaqueOriginIdentifier)
        return std::nullopt;
    if (*opaqueOriginIdentifier)
        return SecurityOriginData(**opaqueOriginIdentifier);
    // decode protocol, host and port
    if (!protocol->isHashTableDeletedValue())
        return std::nullopt;
    return SecurityOriginData(protocol, host, port);
}
```

Reply:
Good point, that's a much better idea. Thanks